public/static/js/spawned_hosts.js

```diff
@@ -134,6 +136,19 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope','$window', '$timeout', 'mciSp
       );
     };
 
+    $scope.fetchAllowedInstanceTypes = function() {
```

Comment:
Instead of making a second request once the page loads, you can send up the allowed instance types with the initial page data

Reply:
Offline: Right now the allowed instance types are provider-specific and stored as part of the config for each provider (currently just AWS). This means that the options to show in the dropdown depend on the provider of the current host (even though spawn hosts now are all EC2). We might want to change how allowed instance types are stored later.